shared/lib/custom_dc_util.py

```diff
@@ -27,6 +30,15 @@ def get_user_data_path() -> str:
   return os.environ.get('USER_DATA_PATH', '')
 
 
+def get_topic_cache_path() -> str:
+  base_path = get_user_data_path()
+  if not base_path:
+    return base_path
+  if is_gcs_path(base_path):
+    return join_gcs_path(base_path, _TOPIC_CACHE_PATH)
+  return os.path.join(base_path, _TOPIC_CACHE_PATH)
+
```

Comment:
Why is this in shared lib when used only by server?

Reply:
To localize knowledge about relative custom dc paths (`datacommons/nl/custom_dc_topic_cache.json`) in this file. Want to move others such as (`datacommons/nl/custom_embeddings.yaml`) and any future ones in this file as well.

If you feel otherwise, I can move it.

Reply:
To me, `shared` implies it can be safely called by multiple servers (nl-server, web-server, model-server), but that's probably not true here?

Reply:
If we should be mindful of something, add a comment on top?  I would additionally point out this is really only for sharing between Web and NL server.

Reply:
Done.